universal/include/userver/logging/json_string.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+/// @file userver/logging/json_string.hpp
+/// @brief @copybrief logging::JsonString
+
+#include <string>
+#include <string_view>
+
+#include <userver/formats/json/value.hpp>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace logging {
+
+/// One line json string.
+class JsonString {
+ public:
+  /// @brief Constructs from provided json object.
+  /// The generated json string is an one line string.
+  JsonString(const formats::json::Value& value);
+
+  /// @brief Constructs from provided json string. It is the user's
+  /// responsibility to ensure that the input json string is valid.
+  /// New lines will be removed during construction.
+  explicit JsonString(std::string json) noexcept;
+
+  /// @brief Constructs "null"
```

Comment:
It does not really, it constructs an invalid json string, as written :thinking:

Perhaps add initializer `json_{"null"};`

Reply:
This comment does make sense together with [the other comment](https://github.com/userver-framework/userver/pull/855#discussion_r1948119986), I promise

Reply:
For an empty string, we simply return "null".
```c++
std::string_view JsonString::Value() const {
  if (json_.empty()) {
    return kNull;
  }
  return json_;
}
```

It helps to avoid allocation for default ctor